dtls.c

```diff
@@ -3676,6 +3676,11 @@ handle_handshake(dtls_context_t *ctx, dtls_peer_t *peer, session_t *session,
   size_t fragment_offset = dtls_uint24_to_int(hs_header->fragment_offset);
 
   if (packet_length > fragment_length){
+    if (!peer || !peer->handshake_params) {
```

Comment:
I'm not sure, which coding style is used, but `!peer` seems to operate on a pointer.

I would prefer 

```c
if (data != NULL) {
```

as also used in that file (around line 1547).

Reply:
`!peer `is used 24 times throughout` dtls.c` as opposed to `peer == NULL`.  Likewise, there a a lot of `if (peer)`.  Line 3679 was copied from line 3756.

Yes, the coding style can be updated throughout, but this will mean a lot of changes.

Reply:
Ah, OK.
I should have spent more time in to investigation before commenting.